### PR TITLE
core: redo search cache from scratch. resolves #10382 resolves #4966 resolves #4088

### LINKED ISSUE #4088: Leaking Memory?
Hey there..
I just discovered Jackett abot a week ago and so far it has been working well. A few days ago i noticed my system getting sluggish took a look and noticed Jackett was using a huge amount of memory. I restarted the service and the memory was relased. I watched as the amount of RAM used increases by the second. after about 12 hours or so Jackett will be using about 10GB of RAM. 
I use it with Medusa, (latest version). I am not sure if Jackett is the problem or Mono. 

Jackett version  0.10.365.0
Mono version 5.16.0.179
System Linux-4.15.0-33-generic-x86_64-with-LinuxMint-18.3-sylvia

[log.txt.20181027.00098-1.txt](https://github.com/Jackett/Jackett/files/2522189/log.txt.20181027.00098-1.txt)

-------------------------------

### LINKED ISSUE #4966: Possible solution for memory issue (?!)
I have a doubt. Sorry if this is a dumb question or if I misunderstood wrong its functioning.
Every time that Sonarr and Radarr request a rss sync or a search, Jackett storage this result. 
If I have a lot of indexers Jackett will consume a lot of memory and not releasing it after the rss sync and, **mainly**, in search.
So, If Jackett clear this cache after Sonarr/Radarr request would not solve this memory issue?

To see this: 
1. look Jackett memory usage
2. do a rss sync in Sonarr or Radarr
3. look again Jackett memory usage
4. add a new indexer in Sonarr or Radarr (or wait some time for indexers get new releases)
5. do a new rss sync in Sonarr or Radarr
6. look Jackett memory usage again

### LINKED ISSUE #10382: core: improve search cache
The current search cache is useless, it's just wasting memory and it's not solving the issue of making less requests.

I have to think a bit but the main idea is to cache `query + response + timestamp`. The same query (indexer + search term + categories..) should return the same response. The only exception is if the indexer return an error. In that case the result is not cached.

Make it configurable:
* MaxAge / TTL => How long the results are in the cache (eg 15 min). That means if you make the same request to Jackett several times you will make just 1 request each 15 minutes.
* Max records => How much results could be in the cache at maximum (limit memory usage)
* Disable cache => To reduce memory usage and for development

Nice to have:
* Show in the response if the results come form the cache or a live request. This can be helpful for debug/development.
* Disable cache if enhanced log is enabled?
* Make configurable cache strategy. You can choose to keep the cache memory as low as possible or keep as many results as possible in the cache (to use the "cache search" feature in the web UI).

Related #7069

### PR DESCRIPTION
**Pull request by ngosang**
In simple words, when you make a request in Jackett, the results are saved in memory (cache). The next request will return results form the cache improving response time and making fewer requests to the sites.
* We assume all indexers/sites are stateless, the same request return the same response. If you change the search term, categories or something in the query Jackett has to make a live request to the indexer.
* There are some situations when we don't want to use the cache:
** When we are testing the indexers => if query.IsTest results are not cached
** When the user updates the configuration of one indexer => We call CleanIndexerCache to remove cached results before testing the configuration
** When there is some error/exception in the indexer => The results are not cached so we can retry in the next request
* We want to limit the memory usage, so we try to remove elements from cache ASAP:
** Each indexer can have a maximum number of results in memory. If the limit is exceeded we remove old results
** Cached results expire after some time
* Users can configure the cache or even disable it

Resolves memory issues and it allows to disable cache: #10382 #4966 #4088

### CODE REVIEW

**Comment by ngosang** [comment]
@garfield69 @ilike2burnthing @xfouloux could you help to test it?

I want to leave it open a few days before merging so that there is some separation with the .net5.0 release and this one. Both PR change important things and it will be hard to track issues.
Should we add some clarifications in the readme/wiki?

**Comment by xfouloux** [comment]
okay, i need to make a new docker image with this branch, i'll loose the cloudflaresolver feature, so i'll do testing for a few days but without some trackers.

**Comment by ilike2burnthing** [comment]
TL;DR - early days, but it works!

Ok, so quick run through the setup here. We're comparing this build against v0.17.10, clean install for each in Windows 10 Sandbox, using 135 indexers (all public, plus Demonoid and TorrentDownload), of which 23 failed (all expected).

In this build TTL was set to 120s, and the default of 1000 was used for max cached results per indexer. The `Test All` function was used (because I'm lazy like that), and all tests were run back to back, unless there is `---` dividing results, in which case there was a ~2.5min gap to run out the cache. The starting baseline is also given (taken ~30s after starting Jackett). Multiple runs were done to get a rough average.

**Test, pause, test, test**

_Cache build_

          WS  Pe  Pr  Sh  Co  PP
    base  82  82  46  36  50  282
    test  141 147 98  43  104 334
    ---
    test  161 161 118 43  125 340
    test  166 171 122 43  128 335

_v0.17.10_

          WS  Pe  Pr  Sh  Co  PP
    base  84  84  48  36  52  282
    test  161 162 118 43  125 336
    ---
    test  180 184 136 43  143 340
    test  205 232 162 43  168 336

**3 successive tests**

_Cache build_

          WS  Pe  Pr  Sh  Co  PP
    base  82  83  47  36  51  282
    test  146 148 103 43  109 340
    test  146 181 103 43  109 337
    test  157 189 113 43  119 337

_v0.17.10_

          WS  Pe  Pr  Sh  Co  PP
    base  81  82  46  36  50  282
    test  164 190 121 43  128 339
    test  176 190 133 43  140 339
    test  197 231 153 43  160 340


<details>
<summary>Legend</summary>

WS- Working Set
Pe - Peak Working Set
Pr - Private Working Set
Sh - Shared Working Set
Co - Commit
PP - Paged Pool
</details>

All tests took ~30s (I had expected cached results to be quicker, but presumably I'm misunderstanding how it works). Removing all failing and slow indexers dropped this to ~20s for all tests, with a slight drop in memory usage.

The main savings seem to be in Private Working Set and Commit, especially for successive runs.

Will do some more testing to confirm there's no issues with disabling cache manually or when enhanced logging is enabled. So far, so good!

**Comment by xfouloux** [comment]
i've done a Dockerfile if someone else wana test 

```
FROM mcr.microsoft.com/dotnet/sdk:latest as build

RUN apt -y update && apt -y install nuget mono-xbuild git
RUN git clone https://github.com/ngosang/Jackett.git && \
 cd Jackett && \
 git checkout feature/cachesup && \
 cd src && \
 dotnet publish Jackett.Server -f net5.0 --self-contained -r linux-x64 -c Debug
ADD ./jackett_launcher.sh /Jackett/src/Jackett.Server/bin/Debug/net5.0/linux-x64/

FROM linuxserver/jackett
RUN rm -rf /app/Jackett
COPY --from=build /Jackett/src/Jackett.Server/bin/Debug/net5.0/linux-x64/ /app/Jackett
```

@ngosang if you have some time, do you think it's possible to merge this branch with @abeloin modification for the cloudflaresolver modification in gui in another branch ? I know it's not in the scope but i would be able to put this cache feature on my seedbox for real live testing.
i'll try to do that on my end but i lack a bit of GIT skills :) Time to learn more ^^

About the cache feature, so far so good, i don't know if it would be great that jackett do a 1st precache of each tracker at startup ? Because right now it's empty unless you query the tracker, but i guess it's fine, even if there is like 1 second difference between 2 same requests (2 sonarr for example) for the same tracker, the 1st query should be cached, and the second from the cache directly ? or the requests are done in parrallel and then in this case it's not cached for the second request because the cache hasn't been done yet via the 1st request ?

**Comment by xfouloux** [comment]
I don't know if it's a bug or something else, but i activated 'Enhanced logging' and the cache stopped working, even now that i disabled it it still won't work, when i click 'view cached release' it is always empty now, even after a search

**Comment by ngosang** [comment]
> EDIT: apparently I can't read - When we are testing the indexers => if query.IsTest results are not cached. So should any of what I did actually have made a difference in memory usage?

@ilike2burnthing Yes, tests are not cached in both cases. The times should be the same and the memory similar. I changed a lot of code so could be some memory improvements too.

> Yea, quick test with torrent-pirat, an annoyingly slow indexer - first search is typically slow (~20s), and successive ones are instantaneous. Running the test for it is as slow each time, and likewise when searching with the cache disabled.

That is the expected behaviour. Could you repeat the first test (measuring memory) but making searches in several indexers (at the same time if you want)? Leave a pause in the middle for the cache to expire. In build v17.10 the cache expires in 1 hour, we will not be able to test it well, but we can see that the memory goes down in this PR.

> @ngosang if you have some time, do you think it's possible to merge this branch with @abeloin modification for the cloudflaresolver modification in gui in another branch ? I know it's not in the scope but i would be able to put this cache feature on my seedbox for real live testing.

Sorry, I don't have time. I think this PR will be merged soon, so you can ask @abeloin to update his Docker image.

> About the cache feature, so far so good, i don't know if it would be great that jackett do a 1st precache of each tracker at startup ?

Warm up the cache is a good practice but in this case I can't know what queries I have to cache. Jackett support a lot of search parameters and little changes in them produce different results. Furthermore the cache expires after some time. I'm going to keep the code as it's for now.

> Because right now it's empty unless you query the tracker, but i guess it's fine, even if there is like 1 second difference between 2 same requests (2 sonarr for example) for the same tracker, the 1st query should be cached, and the second from the cache directly ? or the requests are done in parrallel and then in this case it's not cached for the second request because the cache hasn't been done yet via the 1st request ?

Requests run in parallel. If the first requests ends before the second one starts (less than 1s), the second one will use the results from cache. If not, we are making 2 requests to the site and the cache will be overwritten with the second result. It's expected.

I think Sonarr and Radarr are making different queries because they use different categories to filter tv/movies. If that is true you will have "2 queries in the cache" one for each one because they are different. That is expected too, I can't give tv series to Radarr ;) The test that I find most interesting is to configure your Sonarr and Radarr instance, leave it for a couple of days and see the Jackett logs. We may find out why you are making so many requests.

Another thing I have to fine tune is the default values for cache TTL and Max results per indexer. What do you think? What are the RSS search intervals in Sonarr, Radarr, Readarr... ?

**Comment by xfouloux** [comment]
For now i'm still at 2H RSS intervall for my sonarr1,sonarr2,radarr,lidarr,readarr
i've put 2500 max result per indexer, 2000s TTL

i've sent a pm to generation-free admins telling them i'm testing this jackett beta cache and i want to check with them if it's way better for them, i'll put 1H RSS in a few hours and see what happens.

i don't remember if sonarr/radarr..etc do any search when it trigger task : Refresh Series
if it is, it might be what is making me explode the request counter having 2 sonarr with the same exact tv show lists (1 is for VO, the other is for MULTI) and well this is my sonarr stats :
image
Radarr might be doing the same every once in a while, and i've this on radarr :
image
lidarr has lots too :
image
but because some tracker have small ressources, it's hammering them bad, sometimes it takes more than a minute to load generation-free index page.

In any case it's running on my seedbox, i'm just not having YGG because of cloudflare so it's fine.
i'll tell you in this PR if i find any issue on my usage

Thanks in any case for such a prompt work on the cache part of jackett man !

**Comment by ilike2burnthing** [comment]
Will redo that test shortly, I was just very much in need of sleep at the time.

Sonarr's default is 15mins, and Radarr's is 60mins. I'll check Lidarr and Readarr.

**Comment by ilike2burnthing** [comment]
Similar to above, but using 33 indexers and running keywordless searches and keyword searches for `720p 2020`. Time taken varied this time, hence the additional column.

**Search: keywordless**

_Cache build_

          WS  Pe  Pr  Sh  Co  PP   Time
    base  83  84  47  36  52  284
    test  160 169 117 41  125 302  ~35s
    ---
    test  162 179 121 41  129 302  ~35s
    test  158 179 116 41  124 301  ~25s

_v0.17.11.0_

          WS  Pe  Pr  Sh  Co  PP   Time
    base  84  84  48  36  52  282
    test  169 170 129 40  130 300  ~40s
    ---
    test  180 180 139 41  147 306  ~40s
    test  196 200 156 41  161 306  ~35s

**Search: 720p 2020**

_Cache build_

          WS  Pe  Pr  Sh  Co  PP   Time
    base  83  83  47  36  52  284
    test  153 155 113 40  122 302  ~30s
    ---
    test  157 159 116 41  125 306  ~30s
    test  153 159 110 41  118 305  ~20s

_v0.17.11.0_

          WS  Pe  Pr  Sh  Co  PP   Time
    base  82  82  46  36  51  282
    test  148 149 108 40  117 302  ~30s
    ---
    test  173 173 133 41  141 306  ~30s
    test  184 188 143 41  149 306  ~25s


<details>
<summary>Legend</summary>

WS- Working Set
Pe - Peak Working Set
Pr - Private Working Set
Sh - Shared Working Set
Co - Commit
PP - Paged Pool
</details>

So yea, definitely working!

**Comment by abeloin** [comment]
@xfouloux Better late than never, I'll merge this commit into my jackett branch so you can test it.

**Comment by xfouloux** [comment]
> @xfouloux Better late than never, I'll merge this commit into my jackett branch so you can test it.

Awesome =) ! i try to do it but, i'm a sysadmin, not a dev =p

**Comment by xfouloux** [comment]
> @xfouloux Better late than never, I'll merge this commit into my jackett branch so you can test it.

You might want to make another branch, so i make another docker tag, so everyone is not updated ?
Else we see another opportunity if you use the same branch, everyone that is using my container cloudproxy tag, would become a beta tester  of the cache PR =)

**Comment by abeloin** [comment]
> Awesome =) ! i try to do it but, i'm a sysadmin, not a dev =p

Neither am I ;)

> You might want to make another branch, so i make another docker tag, so everyone is not updated ?
Else we see another opportunity if you use the same branch, everyone that is using my container cloudproxy tag, would become a beta tester of the cache PR =)

Yes I made another branch but looking at your docker script, it is pulling the end results which is a released binary. As far as I know, releases are project based not branch based so it would still be picked up by the docker script since it's only looking for the first "tag_name".

Hence why I had to jump through some hoops to make it work(release 0.117.1 and 0.117.3 are duplicate).

It would be better if the script could look for both "tag_name" and "prerelease":
https://github.com/xfouloux/docker-jackett/blob/cloudproxy/root/etc/cont-init.d/30-config#L8

This way it can skip when a release is a pre-release.

Binaries are available here: https://github.com/abeloin/Jackett/releases/tag/v0.117.2

Have fun!

**Comment by xfouloux** [comment]
> > Awesome =) ! i try to do it but, i'm a sysadmin, not a dev =p
> 
> Neither am I ;)
> 
> > You might want to make another branch, so i make another docker tag, so everyone is not updated ?
> > Else we see another opportunity if you use the same branch, everyone that is using my container cloudproxy tag, would become a beta tester of the cache PR =)
> 
> Yes I made another branch but looking at your docker script, it is pulling the end results which is a released binary. As far as I know, releases are project based not branch based so it would still be picked up by the docker script since it's only looking for the first "tag_name".
> 
> Hence why I had to jump through some hoops to make it work(release 0.117.1 and 0.117.3 are duplicate).
> 
> It would be better if the script could look for both "tag_name" and "prerelease":
> https://github.com/xfouloux/docker-jackett/blob/cloudproxy/root/etc/cont-init.d/30-config#L8
> 
> This way it can skip when a release is a pre-release.
> 
> Binaries are available here: https://github.com/abeloin/Jackett/releases/tag/v0.117.2
> 
> Have fun!

Oh i can do that, i must have a exemple of script somewhere =), i'll work on that soon thanks =)

**Comment by ilike2burnthing** [comment]
@xfouloux look at the Pr and Co columns, and compare the cache build with v0.17.11.0:

cache build - little to no change between searches, with or without pulling from the cache
v0.17.11.0 - steady increase with each search

The cache build is also slightly faster.

**Comment by xfouloux** [comment]
Thanks @abeloin it is done now,

if you use 
```-e JACKETT_PRE_BUILD=1```
when you launch my container (cloudproxy tag) it will pull the latest prerelease instead of the latest release

**Comment by ngosang** [comment]
> Sonarr's default is 15mins, and Radarr's is 60mins. I'll check Lidarr and Readarr.
LazyLibrarian = 20mins
Readarr = 15mins
Lidarr = 15mins

@ilike2burnthing @garfield69  I don't know what will be the best value. The old cache is 1 hour. I'm thinking in 50 minutes (3000) to avoid race conditions. Is it too much? Users will get new torrents with a delay... On average they will get the torrents with 25 min delay, I think it's acceptable for the average user. If someone need to be more aggressive he can change the parameters, even disable the cache,

> Similar to above, but using 33 indexers and running keywordless searches and keyword searches for 720p 2020. Time taken varied this time, hence the additional column.

@ilike2burnthing In your test you are getting ~30 seconds because one or more of the indexers is not working. The query for that indexer is not cached and it's delaying the entire request. If all indexers will work the response should be instantaneous.

I'm thinking in adding another kind of cache skip trackers with errors. Something like Sonarr does. If some tracker returns 2 consecutive errors I can disable that tracker for 15 minutes to avoid slowing other requests. This will be useful for the manual searches in the UI.

> i've put back 30min RSS feed on all my *arr to see what happens

@xfouloux Could you do some test for me? Configure Jackett `TTL=7200` (2 hours) and `Max requests per indexer = 10000`. Leave it running for half a day and share the logs with me.

**Comment by ilike2burnthing** [comment]
Given that the TTL is configurable, I'd say 50mins is fine, and if someone really wants to change it they can.

The 33 indexers I used were my day-to-day public indexers, minus a few that were failing or are unreliable (e.g. EXT, RARBG, Torrentz2). They all tested successfully before starting, and I didn't notice any failing during, though admittedly I wasn't paying close attention to that.

I'll recheck to see if I missed something, as I had expected the results to be relatively instantaneous. If one found 0 results, would that also cause the search to be re-run each time?

> I'm thinking in adding another kind of cache to skip trackers with errors. Something like Sonarr does. If some tracker returns 2 consecutive errors I can disable that tracker for 15 minutes to avoid slowing other requests. This will be useful for the manual searches in the UI.

Sounds good!

**Comment by xfouloux** [comment]
@ngosang i changed my jackett settings like you asked, but hum, is there any way to keep more logs for jackett ?
Because this is what i have :

```
1.6M -rw-rw-r-- 1 seed seed 1.6M Dec 11 01:31 log.txt
2.0M -rw-rw-r-- 1 seed seed 2.0M Dec 10 16:21 log.txt.20201210.00067.txt
2.0M -rw-rw-r-- 1 seed seed 2.0M Dec 10 16:21 log.txt.20201210.00068.txt
2.0M -rw-rw-r-- 1 seed seed 2.0M Dec 10 16:21 log.txt.20201210.00069.txt
2.0M -rw-rw-r-- 1 seed seed 2.0M Dec 10 16:26 log.txt.20201210.00070.txt
2.0M -rw-rw-r-- 1 seed seed 2.0M Dec 10 16:27 log.txt.20201210.00071.txt
```
and it rotates, and i'm not even on enhanced logging

**Comment by ilike2burnthing** [comment]
So it seems to be dependent on the number of returned results and the number of results displayed.

30 indexers

**Search: 720p 2020**
_Show 20 entries_
1st run  - 39s, 2468 entries
2nd run - 20s, 2468 entries

_Show 500 entries_
1st run  - 49s, 2468 entries
2nd run - 43s, 2468 entries

**Search: The.Daily.Show.2020.12.08.720p.WEB.h264-BAE**
_Show All entries_
1st run  - 8s, 31 entries
2nd run - 0s, 31 entries

Some indexers returned 0 results for the last search, and the 2nd run was still instantaneous, so it doesn't seem to be an issue.

**Comment by xfouloux** [comment]
@ngosang i've sent you an email with latest 14hours of logs, but not enhanced

**Comment by ilike2burnthing** [comment]
Not through Jackett, no, sorry, it's set here: https://github.com/Jackett/Jackett/blob/b5a6aab894d45c2d43f71e4db3452e5932d77fa7/src/Jackett.Common/Utils/LoggingSetup.cs#L30

Only thing I can think of would be running a script to monitor the folder and copy the logs on change. Or a lazier approach, just have the logs be copied every 5s, skipping existing files.

**Comment by xfouloux** [comment]
> MaxArchiveFiles

Yeah well i'll do something this week end, thanks =)

**Comment by ngosang** [comment]
@ilike2burnthing I changed my mind. I think 35 minutes is a better default value because some private trackers publish a lot of new torrents each hour. If the cache TTL is high you can miss some releases in the newest page.

@xfouloux Thank you for the logs. The cache is working well for all trackers but TorrentLeech and cpasbien. Also you are making a lot of request in the same minute. Must be something wrong in your installation. For example, you are making 29 requests to cpasbien in 3 minutes.
```
2020-12-11 05:53:46.3240 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:53:51.5737 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:53:56.7903 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:02.3432 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:07.3664 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:12.4177 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:17.4912 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:22.6620 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:27.9229 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:33.0322 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:38.2115 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:43.4300 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:48.4868 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:53.5920 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:54:58.8004 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:04.1273 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:09.1161 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:14.4182 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:19.3932 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:24.6135 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:29.8035 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:34.8043 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:40.2479 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:45.4045 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:50.5277 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:55:55.6687 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:56:00.7591 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:56:05.8033 Info Torznab search in cpasbien => Found 100 releases 
2020-12-11 05:56:10.9447 Info Torznab search in cpasbien => Found 100 releases 
```
Also you have some errors and warnings in your logs related to categories. I recommend you to take a look and change the categories Radarr and Sonarr are using to search in Jackett. It will improve the results you are getting.

Thank you guys. This is ready!

**Comment by xfouloux** [comment]
> @ilike2burnthing I changed my mind. I think 35 minutes is a better default value because some private trackers publish a lot of new torrents each hour. If the cache TTL is high you can miss some releases in the newest page.

I would say it is kinda wrong @ngosang, depending on the tracker implementation of course, the current behavior of *arr is to search for new release until it find the old one, so if the tracker support pagination you're ok, if not yeah you miss torrents
Might be something you wanna put as an info field maybe on the checkbox to activate cache ?

> @xfouloux Thank you for the logs. The cache is working well for all trackers but TorrentLeech and cpasbien. Also you are making a lot of request in the same minute. Must be something wrong in your installation. For example, you are making 29 requests to cpasbien in 3 minutes.

hum doesn't cpabien and torrentleech doing pagination ? as i said earlier ?

Yeah i'm fixing wrong categories sometimes, but i've a lot configured and it's taking me some time =p !
> 
> ```
> 2020-12-11 05:53:46.3240 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:53:51.5737 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:53:56.7903 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:02.3432 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:07.3664 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:12.4177 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:17.4912 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:22.6620 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:27.9229 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:33.0322 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:38.2115 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:43.4300 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:48.4868 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:53.5920 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:54:58.8004 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:04.1273 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:09.1161 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:14.4182 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:19.3932 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:24.6135 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:29.8035 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:34.8043 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:40.2479 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:45.4045 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:50.5277 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:55:55.6687 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:56:00.7591 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:56:05.8033 Info Torznab search in cpasbien => Found 100 releases 
> 2020-12-11 05:56:10.9447 Info Torznab search in cpasbien => Found 100 releases 
> ```
> 
> Also you have some errors and warnings in your logs related to categories. I recommend you to take a look and change the categories Radarr and Sonarr are using to search in Jackett. It will improve the results you are getting.
> 
> Thank you guys. This is ready!

**Comment by ngosang** [comment]
> I would say it is kinda wrong @ngosang, depending on the tracker implementation of course, the current behavior of *arr is to search for new release until it find the latest one it got on latest RSS interval search, so if the tracker support pagination you're ok, if not yeah you miss torrents (unless i've been misinformed all this time =p ???)
Might be something you wanna put as an info field maybe on the checkbox to activate cache ?

Only 3/500 trackers support pagination in Jackett. Neither cpabien or torrentleech support it.  #975

**Comment by xfouloux** [comment]
well i don't have a clue of what is doing that, for me it was when RSS interval reached, it was doing pagination search on indexers to get all results from last time
